core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -96,39 +82,64 @@ public static boolean ingestNode(Settings settings) {
 
     public static final List<DiscoveryNode> EMPTY_LIST = Collections.emptyList();
 
-    private String nodeName = "";
-    private String nodeId;
-    private String hostName;
-    private String hostAddress;
-    private TransportAddress address;
-    private ImmutableOpenMap<String, String> attributes;
-    private Version version = Version.CURRENT;
+    private final String nodeName;
+    private final String nodeId;
+    private final String hostName;
+    private final String hostAddress;
+    private final TransportAddress address;
+    private final ImmutableOpenMap<String, String> attributes;
```

Comment:
I wonder if we really need this ImmutableOpenMap or if we can just use a default one from the JDK?